<h1><center> Suicide Rate among Regions</center></h1> 
__<div style="text-align: right"> Junyan Huang</div>__
__<div style="text-align: right"> Oct 2, 2018 </div>__

### Introduction 
The project is to find out the suicide rate among regions. The tables needed are sui_patient and sui_hospital which are linked by hospitalid. These two tables are suicide related data. 

The first step is to count the number of icu records in each hospital. The medical records are indicated by patientunitstayid. Patientunitstayid is the unique id for each ICU stay. The same patient may have more than one patientunitstayid if he/she has more than one ICU visit record. In the project, such issue is not addressed. The project is just based on the patientunitstayid. The second step is to count the number of ICU records in each regions. And finally, to plot the distribution.

### Extract the number of suicide records in each hospital

In [94]:
import pandas as pd

In [95]:
hospital = pd.read_csv("/home/cs52205/hospital.csv.gz", compression="gzip", header=0, sep=",")
print(hospital.head())
print(hospital.shape)

   hospitalid numbedscategory teachingstatus   region
0          56            <100              f  Midwest
1          58       100 - 249              f  Midwest
2          59            <100              f  Midwest
3          60            <100              f  Midwest
4          61            <100              f  Midwest
(208, 4)


In [111]:
sui_hospital = pd.read_csv("/home/cs52205/suicide_records/sui_hospital.csv", header=0, sep=",")
print(sui_hospital.head())
print(sui_hospital.shape)

   Unnamed: 0  hospitalid numbedscategory teachingstatus   region
0           0          56            <100              f  Midwest
1           1          58       100 - 249              f  Midwest
2           3          60            <100              f  Midwest
3           4          61            <100              f  Midwest
4           5          63       100 - 249              f  Midwest
(168, 5)


In [112]:
not_sui_hospital=pd.read_csv("/home/cs52205/suicide_records/not_sui_hospital.csv", header=0, sep=",")
print(not_sui_hospital.head())
print(not_sui_hospital.shape)

   Unnamed: 0  hospitalid numbedscategory teachingstatus   region
0           2          59            <100              f  Midwest
1          13          83             NaN              f      NaN
2          14          84            <100              f  Midwest
3          15          85             NaN              f      NaN
4          16          86             NaN              f  Midwest
(40, 5)


In [113]:
patient = pd.read_csv("/home/cs52205/patient.csv.gz", compression="gzip", header=0, sep=",", index_col=0)
sui_patient = pd.read_csv("/home/cs52205/suicide_records/sui_patient.csv", header=0, sep=",", index_col=0)
not_sui_patient = pd.read_csv("/home/cs52205/suicide_records/not_sui_patient.csv", header=0, sep=",", index_col=0)

In [114]:
#print(patient.head())

Patient table and hospital table are linked by hospitalid

In [115]:
print(sui_patient.shape)

(1470, 29)


In [116]:
sui_records = sui_patient.groupby("hospitalid")[["patientunitstayid"]].nunique()

In [117]:
sui_records.rename(columns = {'patientunitstayid':'sui_records'}, inplace = True)

In [118]:
sui_records.reset_index(inplace=True)

In [119]:
print(sui_records.head())

   hospitalid  sui_records
0          56            3
1          58           11
2          60            6
3          61            1
4          63            3


There are totally 168 unique hospitals in the table. The number of ICU records are shwon as above. For instance, in hospital id 56, there are 3 ICU records.

### Get the number of suicide records in each region.

In [120]:
print(sui_hospital.head())

   Unnamed: 0  hospitalid numbedscategory teachingstatus   region
0           0          56            <100              f  Midwest
1           1          58       100 - 249              f  Midwest
2           3          60            <100              f  Midwest
3           4          61            <100              f  Midwest
4           5          63       100 - 249              f  Midwest


Since the hospitalid in the sui_hospital and sui_records are the same in both length and order, I will extract the region from sui_hospital table and records from sui_records and make a new dataframe.

In [123]:
region = sui_hospital[["hospitalid",'region']].copy()
print(region.head())

   hospitalid   region
0          56  Midwest
1          58  Midwest
2          60  Midwest
3          61  Midwest
4          63  Midwest


In [127]:
sui_region = pd.concat([region, sui_records], axis=1)
print(sui_region.head())

   hospitalid   region  hospitalid  sui_records
0          56  Midwest          56            3
1          58  Midwest          58           11
2          60  Midwest          60            6
3          61  Midwest          61            1
4          63  Midwest          63            3


In [139]:
sui_records_raw = sui_patient[["patientunitstayid", "hospitalid"]].copy()

In [140]:
#sui_records_raw = sui_records_raw.reset_index(drop=True)

In [141]:
print(sui_records_raw.head())

     patientunitstayid  hospitalid
298             143389          66
524             144741          66
738             146060          69
803             146486          73
904             147041          61
